### ISSUE TITLE
test image resize and count

### ISSUE THREAD

**Issue by faridamousa** [ultralytics/ultralytics#12146]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

when i test yolov8 on an image, the image gets resized first and tells me that no detection occurs. i want to not resize the image before testing. and why would i have no detection occurring when in training and validating, the model was detecting very well.
i also want to perform counting on images and the code provided in documentation is for videos. what can i do?

### Additional

please answer as this is for my bachelor project

**Comment by faridamousa** [comment]
i increased the number of epochs from 10 to 30 and it worked.
my problem now is with the counting

**Comment by faridamousa** [comment]
and ialso want to know how to get precision, recall, mAP50, mAP50-95
i wrote: metrics = model.val(data="config.yaml")
is this correct? and how do i know if the values i have are good or not

**Comment by glenn-jocher** [comment]
Yes, your command to obtain metrics like precision, recall, mAP50, and mAP50-95 using `model.val(data="config.yaml")` is correct provided that `"config.yaml"` correctly refers to your dataset configuration.

After running the validation, you can access the various metrics as follows:
```python
print(metrics.box)
```
This will print out the precision, recall, mAP50, and mAP50-95 values.

Regarding whether these values are good or not, it typically depends on the complexity of your dataset and the specific requirements of your application. In general, higher values are better, with a mAP (mean Average Precision) of over 0.5 (50%) being decent for many applications, and values over 0.75 (75%) considered very good.

Adjusting your model's training parameters or augmenting your dataset might be necessary to improve these scores if they're lower than desired.

**Comment by faridamousa** [comment]
how can i adjust the training parameters? and how can i make count on images?

**Comment by glenn-jocher** [comment]
Hi there! 😊

To adjust training parameters, you can modify the values in your `.yaml` configuration file related to training. For example, to change the learning rate or batch size, you could adjust the `lr0` or `batch` parameters respectively.

For counting objects in images using a trained model, you can simply run predictions and then tally the number of detected objects by class. Here’s a basic Python snippet using a YOLO model:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/model.pt')

# Run prediction on an image
results = model.predict('path/to/image.jpg')

# Count objects
counts = results.pandas().xyxy[0]['name'].value_counts()
print(counts)
```

This will output the count of each class detected in the image. Let me know if you have more questions! 🌟

**Comment by faridamousa** [comment]
if i want get r2, mse, mae, rmse, rmae, rrmse
how can i do that?
